build-logic/src/main/kotlin/TargetCompatibility.kt

```diff
@@ -10,7 +10,13 @@ fun Project.configureJavaAndKotlinCompilers() {
     // For Kotlin JVM projects
     tasks.withType(KotlinCompile::class.java) {
       it.kotlinOptions {
-        freeCompilerArgs = freeCompilerArgs + "-Xopt-in=kotlin.RequiresOptIn"
+        freeCompilerArgs = freeCompilerArgs +
+            "-Xopt-in=kotlin.RequiresOptIn" +
+            // This is a workaround for https://youtrack.jetbrains.com/issue/KT-47000 (fixed in Koltin 1.6.20)
+            // Since we don't use @JvmDefault anywhere, the option has no effect, but suppresses the bogus compiler error
+            // See also https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-default/
+            // See also https://blog.jetbrains.com/kotlin/2020/07/kotlin-1-4-m3-generating-default-methods-in-interfaces/
+            "-Xjvm-default=compatibility"
```

Comment:
Out of curiosity, why `compatibility` and not `enable`?

Reply:
My thinking was we're not using `@JvmDefault` anywhere at the moment, plus we should probably remove this "soonish" when upgrading to 1.6.20.
But if we don't, and we do use `@JvmDefault` at some point, I'm thinking the `compatibility` mode is safer (the doc says "binary compatible")?  Not 100% sure it's really useful since if this happens, the Api Dump should warn us anyway, right?

Reply:
Not sure how much of apiDump can catch that. I did a run of japicmp with and without `-Xjvm-default=compatibility` and it didn't show any difference so I guess it is safe. I'll feel better once we upgrade to 1.6.20 though